### DISCUSSION TITLE
How May We Apply the Research

### DISCUSSION THREAD

**Discussion by DanGould** [peakshift/bitcoin-ux#32] [Q&A]
I'm noticing so much quality research accumulating here that would benefit from application and user feedback loop. I think we can do much of this with @chaincase-app.

What's the best way to integrate the process? At what point is a research point mature enough to be applied? What tools make it easy to move between theory to prototype to release **for this team?** Looking to collaborate and provide value by way of a user base and fast iteration.

Looking for jumping off points beyond discussions or what you think is particularly low-hanging fruit.

**Comment by johnsBeharry** [comment]
@DanGould 

> I'm noticing so much quality research accumulating here that would benefit from application and user feedback loop. I think we can do much of this with @chaincase-app.

which areas in particular are of interest in this repo? would help with some direction on where to start cleaning up. have not done any house keeping for a while.

> At what point is a research point mature enough to be applied?

Good question, but this also depends on your own goals. 

Ultimately we're researching things that would get presented in the [design abstractions](https://github.com/peakshift/bitcoin-ux/tree/master/abstractions):

- transaction builder
- pending operations
- contacts
- wallet-to-wallet communication
- payment requests
- recharging spending wallet / swaps

> What tools make it easy to move between theory to prototype to release for this team?

currently using ionic and javascript for ui prototypes, probably photon at some point but there's no need at the moment, and would probably stick with these as it allows for quick web based demos to be made. [bitcoin contacts](http://bitcoin-contacts.netlify.app) is an example of such prototype.

> Looking for jumping off points beyond discussions or what you think is particularly low-hanging fruit.

So I guess one of the bigger questions with this contacts based coin selection I have is what is the speed in which coins get tainted? Some simulation would be great for this but I have not had the time. If not simulation an implementation on testnet that is able to export a wallet.json would be great!

**Reply by DanGould** [reply]
> which areas in particular are of interest in this repo? would help with some direction on where to start cleaning up. have not done any house keeping for a while.

Tx builder is probably the greatest sticking point right now. We can iterate from simple send to adding complexity

>  If not simulation an implementation on testnet that is able to export a wallet.json would be great!

https://testflight.apple.com/join/Edhp2cXj export the `Wallet.json` from the settings page
